resources/assets/js/topics.js

```diff
@@ -0,0 +1,72 @@
+let topics = null;
+
+export default {
+    all() {
+        if (topics !== null) {
+            return new Promise(function (resolve) {
+                resolve(topics);
+            });
+        }
+
+        return new Promise(function (resolve, reject) {
+            Vue.http.get('topics').then(
+                response => {
+                    topics = response.data;
+                    resolve(topics);
+                },
+                response => {
+                    reject();
+                    console.log('error', response);
+                }
+            );
+        });
+    },
+
+    add(topic) {
+        return new Promise( (resolve, reject) => {
```

Comment:
If this one wasn't wrapped, it would be returning a vue-resource promise which has a different API than ES6 promises.

Since the `all` call has to be wrapped in an ES6 promise to do the memoization stuff, it's probably best for everything async here to return an ES6 promise instead of the vue-resource promises, so you have a consistent API for handling them.